cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -106,6 +107,17 @@ func clientForUserAgentOrDie(config client.Config, userAgent string) *client.Cli
 	return kubeClient
 }
 
+// clientsetForUserAgentOrDie is going to replace clientForUserAgentOrDie
+func clientsetForUserAgentOrDie(config client.Config, userAgent string) clientset.Interface {
```

Comment:
Instead of adding this function in multiple places, could we have a setup like this?

```
func (c client.Config) AddUserAgent(userAgent string) (newConfig client.Config) { /* make copy, set user agent */ }

...

clientset.NewOrDie(config.AddUserAgent("me"))

```